R/xml_log.R

```diff
@@ -0,0 +1,102 @@
+##' @title  return datatable for parse xml data 
+##' @param log the input log file path
+##' @import xml2
+##' @export
+deepstate_xmlog <- function(log){
```

Comment:
overall this function seems too long/complicated... for readability can you please break it up into sub-routines?

Reply:
also maybe change fun name to deepstate_read_valgrind_xml, and argument to log.file or xml.file ? (it is for valgrind xml files, right?)

Reply:
Sure @tdhock I'll break the function into modules and will update the function name.

Yes, it is for XML files(argument should be xml.file).  I'll update it.

Reply:
Updated the function name, argument name and modularized the code